### PR TITLE
[Milight] Use DS annotations, fix static code analysis findings

### PR DESCRIPTION
**Pull request by wborn**
* Use DS annotations
* Fix static code analysis findings

### CODE REVIEW

File: addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightV3RGBW.java [hunk 1]
```diff
@@ -39,11 +38,10 @@ public void setHSB(int hue, int saturation, int brightness, MilightThingState st
         if (saturation < 50) {
             whiteMode(state);
         } else {
-
             state.hue360 = hue;
-            final byte messageBytes[] = new byte[] { 0x40, make_color(hue), 0x55 };
-            final byte c_on[] = { command_on[zone], 0x00, 0x55 };
-            sendQueue.queue(QueueItem.createRepeatable(uidc(CAT_COLOR_SET), c_on).addRepeatable(messageBytes));
+            final byte messageBytes[] = new byte[] { 0x40, makeColor(hue), 0x55 };
+            final byte con[] = { COMMAND_ON[zone], 0x00, 0x55 };
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightV3RGBW.java]: cOn?

File: addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightV3RGBW.java [hunk 2]
```diff
@@ -54,34 +52,32 @@ public void setHSB(int hue, int saturation, int brightness, MilightThingState st
...
         }
     }
 
     @Override
     public void whiteMode(MilightThingState state) {
-        final byte c_on[] = { command_on[zone], 0x00, 0x55 };
-        final byte c_white[] = { command_whitemode[zone], 0x00, 0x55 };
-        sendQueue.queue(QueueItem.createRepeatable(uidc(CAT_WHITEMODE), c_on).addRepeatable(c_white));
+        final byte con[] = { COMMAND_ON[zone], 0x00, 0x55 };
+        final byte cwhite[] = { COMMAND_WHITEMODE[zone], 0x00, 0x55 };
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightV3RGBW.java]: cWhite?

Comment (martinvw) [reply addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightV3RGBW.java]: Etc 😉

File: addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightV3RGBW.java (hunk 1 above)
Comment (wborn) [reply addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightV3RGBW.java]: Fixed

File: addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightV3RGBW.java (hunk 2 above)
Comment (wborn) [reply addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightV3RGBW.java]: Fixed

**Comment by wborn** [comment]
Only now I see there is another Milight PR https://github.com/openhab/openhab2-addons/pull/2752 with some overlap. :hushed:

@davidgraeff are you planning to rebase your PR on the current master to resolve the current merge conflicts so I address the remaining SAT issues in this PR? Or should I add your additional changes in this PR or are you willing to create a new PR based on the SAT issues fixed in this PR?

**Comment by davidgraeff** [comment]
@wborn: Please add necessary changes from my PR to yours :)

**Comment by wborn** [comment]
> @wborn: Please add necessary changes from my PR to yours :)

What still seems to be missing from this PR is an easy way to enable/disable the "Milight V6 iBox bridge emulation". 

We could move that code to the test bundle like @kaikreuzer [suggested](https://github.com/openhab/openhab2-addons/pull/2752#issuecomment-369146060). An issue with that approach could be that the content of `libs-pullrequest-local` is not stable, see [Distributing bindings through the IoT Marketplace](https://community.openhab.org/t/distributing-bindings-through-the-iot-marketplace/24491?u=wborn):

> Note: The shared Maven repo for PRs obviously has the risk that if somebody else includes changes to a certain binding in his PR that the binary jar is changed accordingly - so it can be tricky to tell, what its origin is. I therefore only recommend using these urls for new bindings that are not yet part of the repo - refactorings of existing bindings are probably better offered elsewhere.

There don't seem to be any test bundles in the more stable online snapshot/release repositories:
* https://bintray.com/openhab/mvn/online-repo
* https://openhab.jfrog.io/openhab/online-repo-snapshot

This could be resolved by hosting the bundle at another location. Perhaps the installation of that bundle could be simplified by including it in the IoT Marketplace... however this can become quite a maintenance burden when it has to be done for several openHAB versions.

---

Another approach could be to simply add a configuration parameter to the Milight Binding where you can enable/disable the emulation, e.g.:

<a href="https://user-images.githubusercontent.com/12213581/37488547-5eb6baf6-2895-11e8-8158-621b4e8210fc.png"></a>

It could also become an advanced configuration parameter. WDYT?

**Comment by davidgraeff** [comment]
The last option was my original intention. Kai does not agree with this approach, because the emulation part is basically for testing and shouldn't bother the user and shouldn't be installed by default. I think he is right and the emulation service should be moved to the test bundle.
But this should not need to be part of this PR :)

**Comment by wborn** [comment]
> But this should not need to be part of this PR :)

That's OK with me @davidgraeff since it seems to be non-trivial to move it to a test bundle. When I moved the emulation bridge to a test bundle it wouldn't start when installed in an OH instance. Perhaps the packaging of a test bundle with `eclipse-test-plugin` is wrong? Other fragments use `eclipse-plugin` as packaging.

>  If nothing else is missing we should merge this PR asap

That's the only part missing from your PR so if everyone is OK with skipping that part it can be reviewed/merged.

**Comment by davidgraeff** [comment]
@martinvw Can you have a look? I'm happy so far.

**Comment by martinvw** [comment]
> Do you have any other review comments @martinvw? :-)

I fell asleep ;-)

I though it was waiting for something else sorry :-)

**Comment by wborn** [comment]
Many thanks @martinvw! Now @davidgraeff  can address #3046. :-)

**Pull request merged**